packages/react-components/semantic-style-hooks-preview/library/src/component-styles/Search/useSemanticSearchStyles.styles.ts

```diff
@@ -0,0 +1,173 @@
+        '::-webkit-search-decoration': {
+            display: 'none',
+        },
+        '::-webkit-search-cancel-button': {
+            display: 'none',
+        },
+    },
+
+    unfocusedNoContentAfter: {
+        paddingRight: 0,
+    },
+});
+
+const useInputStyles = makeStyles({
+    small: {
+        paddingRight: tokens.spacingHorizontalSNudge,
+    },
+    medium: {
+        paddingRight: tokens.spacingHorizontalS,
+    },
+    large: {
+        paddingRight: tokens.spacingHorizontalMNudge,
+    },
+});
+
+const useContentAfterStyles = makeStyles({
+    contentAfter: {
+        paddingLeft: tokens.spacingHorizontalM,
+        columnGap: tokens.spacingHorizontalXS,
+    },
+    rest: {
+        height: 0,
+        width: 0,
+        paddingLeft: 0,
+        overflow: 'hidden',
+    },
+});
+
+const useDismissClassName = makeResetStyles({
+    boxSizing: 'border-box',
+    color: semanticTokens.foregroundCtrlIconOnNeutralRest, // "icon color" in design spec
+    display: 'flex',
+    // special case styling for icons (most common case) to ensure they're centered vertically
+    // size: medium (default)
+    '> svg': { fontSize: '20px' },
+    cursor: 'pointer',
+});
+
+const useDismissStyles = makeStyles({
+    disabled: {
+        color: semanticTokens.backgroundCtrlSubtleRest,
+    },
+    // Ensure resizable icons show up with the proper font size
+    small: {
+        '> svg': { fontSize: '16px' },
+    },
+    medium: {
+        // included in useDismissClassName
+    },
+    large: {
+        '> svg': { fontSize: '24px' },
+    },
+});
+
+/**
+ * Apply styling to the SearchBox slots based on the state
+ */
+export const useSemanticSearchBoxStyles = (_state: unknown): SearchBoxState => {
+    'use no memo';
+
+    const state = _state as SearchBoxState;
+
+    const { disabled, focused, size } = state;
+
+    const rootStyles = useRootStyles();
+    const inputStyles = useInputStyles();
+    const contentAfterStyles = useContentAfterStyles();
+    const dismissClassName = useDismissClassName();
+    const dismissStyles = useDismissStyles();
+
+    state.root.className = mergeClasses(
+        state.root.className,
+        searchBoxClassNames.root,
+        rootStyles[size],
+        !focused && rootStyles.unfocusedNoContentAfter,
+        getSlotClassNameProp_unstable(state.root),
+    );
+    state.input.className = mergeClasses(
+        searchBoxClassNames.input,
+        rootStyles.input,
+        !focused && inputStyles[size],
+        state.input.className,
+    );
+
+    if (state.dismiss) {
+        state.dismiss.className = mergeClasses(
+            searchBoxClassNames.dismiss,
+            dismissClassName,
+            disabled && dismissStyles.disabled,
+            dismissStyles[size],
+
+            state.dismiss.className,
+        );
+    }
+
+    if (state.contentBefore) {
+        state.contentBefore.className = mergeClasses(searchBoxClassNames.contentBefore, state.contentBefore.className);
```

Comment:
getSlotClassNameProp_unstable and move state.contentBefore.className to first param